content/intro-to-onchain.md

```diff
@@ -0,0 +1,96 @@
+ - checking the accounts provided with incoming instructions, for example to check that certain accounts are of a particular type, or are distinct from other accounts.
+
+Regardless of the language and framework you choose, Solana works the same. Let's refresh how programs work on Solana.
+
+![Diagram showing a transaction with twp instructions](../assets/transaction-and-instructions.svg)
+
+## Programs are deployed at addresses
+
+Like like we can send tokens to users using their public key, we can find programs using the program's public key. When using Anchor, a keypair is created during `anchor init` and the private key saved in the `target/deploy` directory of your project.
+
+A program's public key is sometimes called a 'program ID'.
+
+## Programs have instruction handlers
+
+For example, a Solana client making a transaction transferring some USDC with a memo saying 'thanks' would have two instructions:
+  - one instruction for the Token program's
+`transfer` instruction handler
+  - the other instruction for the Memo program's `memo` instruction handler. 
+
+Both these instructions must complete successfully for the transaction to execute.
+
+Instruction handlers are how blockchain programs process the instructions from clients. Every exchange, lending protocol, escrow, oracle, etc. provides their functionality by instruction handlers. 
+
+## Instruction handlers write their state to Solana accounts
+
+If you have previously done web development, you can think of instruction handlers like an HTTP route handler, and incoming instructions like HTTP requests. 
+
+But unlike HTTP route handlers, Solana instruction handlers don't return data. Instead, the instruction handlers write their data to accounts on Solana.
+
+Programs on Solana can transfer tokens, which end up in user wallet addresses (for SOL) for the users token accounts (for other tokens). 
+
+But more importantly, programs on Solana can create additional addresses as needed, to store items of data. 
+
+## Programs store data in Program Derived Addresses (PDAs), a key-value store
+
+These addresses are called **program derived addresses** and act as a **key / value store**.
+
+ - The 'key' is the address of the PDA, which is determined by `seeds` chosen by you, the programmer. 
+   - Want an account to store USD to AUD exchange rate? Your program can use the seeds `USD` and `AUD` to make a Program Derived Address
+   - Want to store information about the relationship of two users? You can use **both those users wallet addreesses** as seeds to make a PDA to store that information.
+   - Want an account to store Steve's review of Titanic? Your program can use Steve's **wallet address** and the string `titanic` (or maybe the IMDB ID if you prefer) to make a Program Derived Address.
+   - Want some global infomation for your entire program? You can use a string like `config`. Your program's PDAs are unique, so these won't conflict with other users. 
+ - The value is the data inside.
```

Comment:
Maybe for clarify we say "The value is the data inside the account at the given address" ?

Reply:
Idk this is a weird one actually where we definitely say "it's in the PDA" even though technically PDA stands for an address not an account...idk I'm not opinionated here actually